src/Features/Core/Portable/Completion/CompletionOptions.cs

```diff
@@ -55,8 +56,14 @@ public bool ShouldShowItemsFromUnimportedNamespaces()
         /// This takes into consideration the experiment we are running in addition to the value
         /// from user facing options.
         /// </summary>
-        public bool ShouldShowNewSnippetExperience()
+        public bool ShouldShowNewSnippetExperience(Document document)
         {
+            // Will be removed once semantic snippets will be added to razor.
+            if (document.IsRazorDocument())
```

Comment:
@ToddGrun asked in [the old PR](https://github.com/dotnet/roslyn/pull/64377#discussion_r984096545) if this works for legacy Razor files, and no, it wouldn't. This just checks for LSP stuff.

@jasonmalinowski do you know if there is an equivalent for checking if the document is a projection buffer in the legacy editor? Would checking filepath work, or do those get faked to `.cs`?

Reply:
I think the file path might work, but the more robust option might be to callhttps://github.com/dotnet/roslyn/blob/5adb79fef405a34d375f7d8fe7aa2e78ea33971a/src/VisualStudio/Core/Def/Venus/ContainedDocument.cs#L71; although we probably have some existing interface you can call that through since that lives at the VS layer.